src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/MemberExpansion.cs

```diff
@@ -437,24 +459,34 @@ internal StaticMembersExpansion(Type declaredType, Expansion members)
             MemberAndDeclarationInfo member,
             DkmClrValue memberValue,
             EvalResultDataItem parent,
+            DynamicFlagsMap dynamicFlagsMap,
             ExpansionFlags flags)
         {
-            var formatter = resultProvider.Formatter;
+            var declaredType = member.Type;
+            var declaredTypeInfo = dynamicFlagsMap.SubstituteDynamicFlags(member.OriginalDefinitionType, member.TypeInfo as bool[]);
```

Comment:
I believe it would work, but it would have the disadvantage of storing the same map on every member of a given type.  I'm not sure I see a compensating advantage.